app/src/main/java/org/sopt/dosopttemplate/presentation/mainhome/MainHomeActivity.kt

```diff
@@ -4,32 +4,35 @@ import androidx.appcompat.app.AppCompatActivity
 import android.os.Bundle
 import android.widget.Toast
 import androidx.fragment.app.Fragment
-import org.sopt.dosopttemplate.MyApplication
+import androidx.lifecycle.Observer
+import androidx.lifecycle.ViewModelProvider
 import org.sopt.dosopttemplate.R
-import org.sopt.dosopttemplate.module.ServicePool.authService
 import org.sopt.dosopttemplate.databinding.ActivityMainhomeBinding
-import org.sopt.dosopttemplate.data.dto.follower.retrofit2data.RequestInquiryDto
-import org.sopt.dosopttemplate.data.dto.follower.retrofit2data.ResponseInquiryDto
-import org.sopt.dosopttemplate.presentation.doandroid.DoAndroidFragment
-import org.sopt.dosopttemplate.presentation.follower.FollowerFragment
 import org.sopt.dosopttemplate.presentation.home.HomeFragment
-import org.sopt.dosopttemplate.presentation.mypage.MyPageFragment
-import retrofit2.Call
-import retrofit2.Response
 
 
 class MainHomeActivity : AppCompatActivity() {
     private lateinit var binding: ActivityMainhomeBinding
+    private lateinit var mainHomeViewModel: MainHomeViewModel
 
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
         binding = ActivityMainhomeBinding.inflate(layoutInflater)
         setContentView(binding.root)
-
+        mainHomeViewModel = ViewModelProvider(this).get(MainHomeViewModel::class.java)
         //intent로 StringArrayList 형태로 받기
 //        var receivedUserInfoList = intent.getStringArrayListExtra("userInfoList")!!
         //setUserInfoPrefs(receivedUserInfoList)
 
+
+        mainHomeViewModel.navigateTo.observe(this, Observer { fragment ->
+            replaceFragment(fragment)
+        })
```

Comment:
```suggestion
        mainHomeViewModel.navigateTo.observe(this){
            replaceFragment(it)
        })
```

이런 식으로 간결하게 쓸 수 도 있을것 같아요!

Reply:
근데 아래 errorMsg 옵저빙 하는 코드와 같이 있으니 그대로 쓰는게 직관적이긴 합니다!

Reply:
트레일링 람다!